### PR TITLE
comment-syntax: Correct the note about `--comments`

### PR DESCRIPTION
**Pull request by dveeden**
### What is changed, added or deleted? (Required)

With the 8.0.23 MySQL client TiDB specific comments are cleared out, so
even with the latest client you still need to add `--comments`.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- Other reference link(s): https://github.com/pingcap/tidb/issues/24176#issuecomment-823835619

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: comment-syntax.md [hunk 1]
```diff
@@ -150,6 +150,6 @@ For details about the optimizer hints that TiDB supports, see [Optimizer hints](
 
 > **Note**
 >
-> In MySQL client before 5.7.7, TiDB specific comment syntax and optimizer comment syntax are treated as comments and cleared by default. To use the two syntaxes in the old client, add the `--comments` option when you start the client. For example, `mysql -h 127.0.0.1 -P 4000 -uroot --comments`.
+> In MySQL client TiDB specific comment syntax is treated as comments and cleared by default. In MySQL client before 5.7.7 hints are also seen as command and cleared by default. It is recommended to use the `--comments` option when you start the client. For example, `mysql -h 127.0.0.1 -P 4000 -uroot --comments`.
```
Comment (TomShawn) [inline comment-syntax.md:153]:
```suggestion
> In MySQL client, the TiDB-specific comment syntax is treated as comments and cleared by default. In MySQL client before 5.7.7, hints are also seen as commands and cleared by default. It is recommended to use the `--comments` option when you start the client. For example, `mysql -h 127.0.0.1 -P 4000 -uroot --comments`.
```

Comment (dveeden) [reply comment-syntax.md]: I've updated the PR based on this suggested change. Thanks.

**Pull request merged**